Sources/Algorithms/Replace.swift

```diff
@@ -0,0 +1,233 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// replacingOccurrences(of:with:)
+//===----------------------------------------------------------------------===//
+
+extension RangeReplaceableCollection {
+  @inlinable
+  public func replacingOccurrences<Target: Collection, Replacement: Collection>(
+    of target: Target,
+    with replacement: Replacement,
+    subrange: Range<Index>,
+    maxReplacements: Int = .max,
+    by areEquivalent: (Element, Target.Element) throws -> Bool
+  ) rethrows -> Self where Replacement.Element == Element {
+    precondition(maxReplacements >= 0)
+    precondition(!target.isEmpty)
```

Comment:
Couldn't using an empty `target` be sanely defined?  The `replacement` would be inserted between every pair of elements in the affected sub-sequence of `self`, plus before the first and after the last elements.  (If the sub-sequence is empty, the replacement is added only once.)